### PR TITLE
Remove the DB related modules

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Remove the DB related maven modules

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22710

### How to test this PR?

N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by tolusha — APPROVED** [review]

**Comment by tolusha** [comment]
Please test upgrade from the previous che-server to a new one.

**Review by ibuziuk — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
@vinokurig: I have faced 0 workpaces in dashboard following next test scenario
1) Create namespace "**admin-devspaces**".
2) Deploy _Eclipse Che Next_.
3) Open _Eclipse Che Dashboard_ and create several workspaces.
4) Ensure workspaces have deployed to "**admin-devspaces**" namespace.
5) Change Eclipse Che CheCluster custom resource:
```
spec:
  components:
    cheServer:
      debug: false
      deployment:
        containers:
          - image: 'quay.io/eclipse/che-server:pr-630'
            imagePullPolicy: Always
```
6) Wait on "**che-server**" pod re-deployment.
7) Open Eclipse Che Dashboard.

_Expected result_: 3 workspaces showed in Dashboard

_Actual result_: 
- 0 workspaces showed in Dashboard
- `admin-che` namespaces has been created with 0 workspaces
- New workspace has been created in `admin-che` namespace.

Test instance: https://eclipse-che.apps.ocp414-dnochev.crw-qe.com/

**Comment by ibuziuk** [comment]
@vinokurig looks like there are some conflicts now to be resolved

**Comment by vinokurig** [comment]
@dmytro-ndp 
> I have faced 0 workpaces in dashboard instead of 3 following next test scenario ...

I could not reproduce the issue with the rosa cluster and RHPDS cluster.
I've noticed an interesting case though:
1. Deploy a new clean openshift cluster (I used cluster bot or RHPDS).
2. Deploy devspaces to the cluster and login to dashboard.
3. Delete the devspaces: `dsc server:delete`. Wait until `openshift-devspaces` namespace is deleted, but the user namespace e.g. `cluster-admin-devspaces` is present.
4. Deploy Che instance with the original che-server image e.g. `chectl server:deploy -p=openshift -i=quay.io/eclipse/che-server:next` and log-in to dashboard.
5. See: the user namespace of the Che instance is located in the previous user namespace from devspaces: `cluster-admin-devspaces`
6. Edit the custom resources yaml to use `quay.io/eclipse/che-server:pr-630` image for the che-server pod.
7. Wait until the che-server pod is reconciled and check the user namespace.
8. See: new conventional user namespace is created: `cluster-admin-che`

Not sure if we need to create a bug ticket for this case.

**Comment by tolusha** [comment]
@vinokurig 
That's expected behavior.
The template for user's namespace is the following: `<username>-che` for Che and `<username>-devsapces` for DS.

**Comment by dmytro-ndp** [comment]
@vinokurig : it seems that "_che-server_" from PR is ignoring existed user namespaces with next metadata in namespace yaml:
```
  labels:
    app.kubernetes.io/component: workspaces-namespace
    app.kubernetes.io/part-of: che.eclipse.org
  annotations:
    che.eclipse.org/username: admin
```

I have updated step 1 of test scenario in the comment https://github.com/eclipse-che/che-server/pull/630#issuecomment-1893829231

Reference: https://drive.google.com/file/d/1ad56GbRIHYwm7uTrRgjiLCCWn78Y53wK/view

File: wsmaster/che-core-api-system/src/main/java/org/eclipse/che/api/system/server/SystemManager.java
Comment (vinokurig) [reply wsmaster/che-core-api-system/src/main/java/org/eclipse/che/api/system/server/SystemManager.java]: rolled back

**Comment by vinokurig** [comment]
@dmytro-ndp 
>  it seems that "che-server" from PR is ignoring existed user namespaces with next metadata in namespace yaml:

I've fixed that in my latest commit

**Comment by dmytro-ndp** [comment]
@vinokurig : I have checked upgrade scenario described in the comment above https://github.com/eclipse-che/che-server/pull/630#issuecomment-1893829231 and didn't reproduce it - workspaces from `devspaces-admin` namespace were displayed after upgrade che-server to the image 'quay.io/eclipse/che-server:pr-630'.

**Comment by ibuziuk** [comment]
@artaleks9 @dmytro-ndp folks, could you please take a look why PR checks are keep failing?

**Review by ibuziuk — APPROVED** [review]
Verified, but please merge only once we have approval from @dmytro-ndp and all green PR checks

**Comment by dmytro-ndp** [comment]
@vinokurig : I have got another error `Cannot invoke "javax.persistence.EntityManagerFactory.createEntityManager()" because "this.emFactory" is null` when ran user namespace removal scenario.

### Test scenario
1. Install Eclipse Che Next with **quay.io/eclipse/che-server:pr-630** image of che-server deployment
2. Open Eclipse Che Dashboard > User Preferences
3. Add items to `Container Registries`, `Personal Access Tokens`, `SSH Keys` tabs
4. Go to OpenShift Console and remove `<user>-che` namespace
5. Return to Eclipse Che Dashboard and re-login

### Expected result
Eclipse Che Dashboard opened successfully, with empty `Container Registries`, `Personal Access Tokens`, `SSH Keys` tabs.

### Actual result
Dashboard shown error page:
> HTTP Error code 500. Endpoint which throws an error https://eclipse-che.apps.ocp414-dnochev.crw-qe.com/api/kubernetes/namespace/provision. Check the browser logs message.

Screenshot from 2024-02-06 18-55-05

<details><summary>Error log from che-server pod</summary>
<p>

```
2024-02-06 18:15:04,904[io-8080-exec-10]  [ERROR] [c.a.c.r.RuntimeExceptionMapper 55]   - Cannot invoke "javax.persistence.EntityManagerFactory.createEntityManager()" because "this.emFactory" is null
java.lang.NullPointerException: Cannot invoke "javax.persistence.EntityManagerFactory.createEntityManager()" because "this.emFactory" is null
	at com.google.inject.persist.jpa.JpaPersistService.begin(JpaPersistService.java:78)
	at com.google.inject.persist.jpa.JpaLocalTxnInterceptor.invoke(JpaLocalTxnInterceptor.java:47)
	at org.eclipse.che.api.user.server.PreferenceManager.find(PreferenceManager.java:98)
	at org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesNamespaceFactory.getPreferencesNamespaceName(KubernetesNamespaceFactory.java:646)
	at org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesNamespaceFactory.findStoredNamespace(KubernetesNamespaceFactory.java:450)
	at org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesNamespaceFactory.lambda$evaluateNamespaceName$2(KubernetesNamespaceFactory.java:406)
	at java.base/java.util.Optional.or(Optional.java:313)
	at org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesNamespaceFactory.evaluateNamespaceName(KubernetesNamespaceFactory.java:406)
	at org.eclipse.che.workspace.infrastructure.kubernetes.provision.NamespaceProvisioner.provision(NamespaceProvisioner.java:47)
	at org.eclipse.che.workspace.infrastructure.kubernetes.api.server.KubernetesNamespaceService.provision(KubernetesNamespaceService.java:95)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.everrest.core.impl.method.DefaultMethodInvoker.invokeMethod(DefaultMethodInvoker.java:174)
	at org.everrest.core.impl.method.DefaultMethodInvoker.invokeMethod(DefaultMethodInvoker.java:61)
	at org.everrest.core.impl.RequestDispatcher.doInvokeResource(RequestDispatcher.java:329)
	at org.everrest.core.impl.RequestDispatcher.invokeSubResourceMethod(RequestDispatcher.java:319)
	at org.everrest.core.impl.RequestDispatcher.dispatch(RequestDispatcher.java:257)
	at org.everrest.core.impl.RequestDispatcher.dispatch(RequestDispatcher.java:131)
	at org.everrest.core.impl.RequestHandlerImpl.handleRequest(RequestHandlerImpl.java:61)
	at org.everrest.core.impl.EverrestProcessor.process(EverrestProcessor.java:130)
	at org.everrest.core.servlet.EverrestServlet.service(EverrestServlet.java:62)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:631)
	at com.google.inject.servlet.ServletDefinition.doServiceImpl(ServletDefinition.java:290)
	at com.google.inject.servlet.ServletDefinition.doService(ServletDefinition.java:280)
	at com.google.inject.servlet.ServletDefinition.service(ServletDefinition.java:184)
	at com.google.inject.servlet.ManagedServletPipeline.service(ManagedServletPipeline.java:89)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:85)
	at org.eclipse.che.core.metrics.ApiResponseMetricFilter.doFilter(ApiResponseMetricFilter.java:46)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	at org.eclipse.che.multiuser.api.authentication.commons.filter.MultiUserEnvironmentInitializationFilter.doFilter(MultiUserEnvironmentInitializationFilter.java:161)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	at org.eclipse.che.commons.logback.filter.RequestIdLoggerFilter.doFilter(RequestIdLoggerFilter.java:50)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	at com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:121)
	at com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:133)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:166)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
	at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:738)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:676)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:357)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:400)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:894)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1741)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

</p>
</details> 

Screencast: 
[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2024.02.06-20_11_55.webm](https://github.com/eclipse-che/che-server/assets/1197777/0d5f078d-d9ff-43b5-904a-ae871adfe8a6)

Eclipse Che pod redeployment didn't help: 
[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2024.02.06-20_14_49.webm](https://github.com/eclipse-che/che-server/assets/1197777/093b59bb-ae52-4fd6-9267-b99ae001ab33)

Che server logs: 
[che-f4fdf6589-mn2nq-che.log](https://github.com/eclipse-che/che-server/files/14184786/che-f4fdf6589-mn2nq-che.log)

**_Note_**: the error wasn't reproduced with **quay.io/eclipse/che-server:next**

**Comment by vinokurig** [comment]
@dmytro-ndp 
> I have got another error Cannot invoke "javax.persistence.EntityManagerFactory.createEntityManager()" because "this.emFactory" is null when ran user namespace removal scenario.

Thank you for the testing, I've fixed the bug. Please check the PR again when the PR image is ready.

**Comment by dmytro-ndp** [comment]
@vinokurig : I haven't reproduced the issue when [ran user namespace removal scenario](https://github.com/eclipse-che/che-server/pull/630#issuecomment-1930554502) again with fixed quay.io/eclipse/che-server:pr-630
Thank you for a quick fix!

**Comment by ibuziuk** [comment]
It was decided not to merge the PR in the 7.82.0, and get it merged for 7.83.0
Before merging, we need to check if API exposed by swagger works as expected

**Review by dmytro-ndp — APPROVED** [review]
looks good to merge, IMHO

**Comment by vinokurig** [comment]
@ibuziuk 
> Before merging, we need to check if API exposed by swagger works as expected
Checked the swagger requests, all works as expected.

**Pull request merged**
